pySDC/implementations/problem_classes/HeatEquation_Chebychev.py

```diff
@@ -0,0 +1,350 @@
+            + (self.b - self.a) / 2 * self.x
+            + (self.b + self.a) / 2
+        )
+        u[iux] = (
+            self.f * np.pi * xp.cos(self.f * np.pi * self.x) * xp.exp(-self.nu * (self.f * np.pi) ** 2 * t)
+            + (self.b - self.a) / 2
+        )
+
+        if noise > 0:
+            assert t == 0
+            _noise = self.u_init
+            rng = self.xp.random.default_rng(seed=666)
+            _noise[iu] = rng.normal(size=u[iu].shape)
+            noise_hat = self.transform(_noise)
+            low_pass = self.get_filter_matrix(axis=0, kmax=self.nvars - 2)
+            noise_hat[iu] = (low_pass @ noise_hat[iu].flatten()).reshape(noise_hat[iu].shape)
+            _noise[:] = self.itransform(noise_hat)
+            u += _noise * noise * (self.x - 1) * (self.x + 1)
+
+        self.check_BCs(u)
+
+        if self.spectral_space:
+            u_hat = self.u_init
+            u_hat[...] = self.transform(u)
+            return u_hat
+        else:
+            return u
+
+
+class Heat1DUltraspherical(GenericSpectralLinear):
+    dtype_u = mesh
+    dtype_f = mesh
+
+    def __init__(self, nvars=128, a=0, b=0, f=1, nu=1.0, **kwargs):
+        self._makeAttributeAndRegister(*locals().keys(), localVars=locals(), readOnly=True)
+
+        bases = [{'base': 'ultraspherical', 'N': nvars}]
+        components = ['u']
+
+        GenericSpectralLinear.__init__(self, bases, components, real_spectral_coefficients=True, **kwargs)
+
+        self.x = self.get_grid()[0]
+
+        I = self.get_Id()
+        Dxx = self.get_differentiation_matrix(axes=(0,), p=2)
+
+        S2 = self.get_basis_change_matrix(p_in=2, p_out=0)
+        U2 = self.get_basis_change_matrix(p_in=0, p_out=2)
+
+        self.Dxx = S2 @ Dxx
+
+        L_lhs = {
+            'u': {'u': -nu * Dxx},
+        }
+        self.setup_L(L_lhs)
+
+        M_lhs = {'u': {'u': U2 @ I}}
+        self.setup_M(M_lhs)
+
+        self.add_BC(component='u', equation='u', axis=0, x=-1, v=a, kind="Dirichlet", line=-1)
+        self.add_BC(component='u', equation='u', axis=0, x=1, v=b, kind="Dirichlet", line=-2)
+        self.setup_BCs()
+
+    def eval_f(self, u, *args, **kwargs):
+        f = self.f_init
+        iu = self.index('u')
+
+        if self.spectral_space:
+            u_hat = u.copy()
+        else:
+            u_hat = self.transform(u)
+
+        u_hat[iu] = (self.nu * (self.Dxx @ u_hat[iu].flatten())).reshape(u_hat[iu].shape)
+
+        if self.spectral_space:
+            me = u_hat
+        else:
+            me = self.itransform(u_hat).real
+
+        f[iu][...] = me[iu]
+        return f
+
+    def u_exact(self, t, noise=0):
+        xp = self.xp
+        iu = self.index('u')
+        u = self.spectral.u_init
+
+        u[iu] = (
+            xp.sin(self.f * np.pi * self.x) * xp.exp(-self.nu * (self.f * np.pi) ** 2 * t)
+            + (self.b - self.a) / 2 * self.x
+            + (self.b + self.a) / 2
+        )
+
+        if noise > 0:
```

Comment:
Noise addition code seems duplicated across the Head1D classes ... would be worth putting it in one unique utility function/method somewhere ...

In the same time, would cost a lot to add a possibility to change the seed